backend/src/test/java/com/example/backend/appuser/AppUserControllerTest.java

```diff
@@ -0,0 +1,189 @@
+                        MockMvcResultMatchers.content().json(response));
+
+        this.mvc.perform(post("/api/appuser")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(request))
+                .andExpectAll(
+                        MockMvcResultMatchers.status().isConflict());
+
+    }
+
+
+    @Test
+    @WithMockUser(username = "Test User", roles = "BASIC")
+    void post_LoginWithRegisteredUser() throws Exception {
+
+        /*
+        AppUserRepository appUserRepository = mock(AppUserRepository.class);
+        when(appUserRepository.findByUsername("Test User"))
+                .thenReturn(
+                        Optional.of(new AppUser(
+                                "Test ID",
+                                "Test User",
+                                "Test Password",
+                                "BASIC")));
+
+         */
+
+
+
+        String request = """
+                {
+                    "username": "Test User",
+                    "password": "Test Password"
+                }
+                """;
+        /*
+        String response = """
+                {
+                    "username": "Test User",
+                    "password": "",
+                    "role": "BASIC"
+                }
+                """;
+
+         */
+
+        this.mvc.perform(post("/api/appuser/login")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(request))
+                .andExpect(status().isOk());
+        /*
+                .andExpect(content().json(
+                            """
+                            {
+                                "username": "Test User",
+                                "password": "",
+                                "role": "BASIC"
+                            }
+                            """));
+                            */
+
+
+    }
+
+    @Test
+    @WithMockUser(username = "Test User", password = "Test Password", roles = "BASIC")
+    void getMe_LoggedInWithRegisteredUser() throws Exception {
+        /*
+        AppUserRepository appUserRepository = mock(AppUserRepository.class);
+        when(appUserRepository.findByUsername("Test User"))
+                .thenReturn(
+                        Optional.of(new AppUser(
+                                "Test ID",
+                                "Test User",
+                                "Test Password",
+                                "BASIC")));
+
+
+
+        String response = """
+                    {
+                    "username": "Test User",
+                    "password": "",
+                    "role": "BASIC"
+                    }
+                """;
+
+         */
+
+        this.mvc.perform(get("/api/appuser/me"))
+                .andExpectAll(
+                        MockMvcResultMatchers.status().isOk());
+
+       //                 MockMvcResultMatchers.content().json(response));
```

Comment:
willst du hier nicht auch testen, dass die richtigen daten zurückkommen?

Reply:
Of course! Hier ist jetzt auch noch die @Autowired AppUserRepo und ein save eines Test Users in der Repo dazugekommen
_Changed_